pyGHDL/dom/Expression.py

```diff
@@ -350,3 +381,46 @@ def __init__(self, left: Expression, right: Expression):
         super().__init__()
         self._leftOperand = left
         self._rightOperand = right
+
+
+@export
+class Aggregate(VHDLModel_Aggregate):
+    def __init__(self, elements: List[AggregateElement]):
+        super().__init__()
+        self._elements = elements
+
+    @classmethod
+    def parse(cls, node):
+        from pyGHDL.dom._Translate import GetExpressionFromNode
+
+        choices = []
+
+        choicesChain = nodes.Get_Association_Choices_Chain(node)
+        for item in utils.chain_iter(choicesChain):
+            kind = GetIirKindOfNode(item)
+            if kind == nodes.Iir_Kind.Choice_By_None:
+                value = GetExpressionFromNode(nodes.Get_Associated_Expr(item))
+                choices.append(SimpleAggregateElement(value))
+            elif kind == nodes.Iir_Kind.Choice_By_Expression:
+                index = GetExpressionFromNode(nodes.Get_Choice_Expression(item))
+                value = GetExpressionFromNode(nodes.Get_Associated_Expr(item))
+                choices.append(IndexedAggregateElement(index, value))
+            elif kind == nodes.Iir_Kind.Choice_By_Range:
+                r = GetExpressionFromNode(nodes.Get_Choice_Range(item))
+                value = GetExpressionFromNode(nodes.Get_Associated_Expr(item))
+                choices.append(RangedAggregateElement(r, value))
+            elif kind == nodes.Iir_Kind.Choice_By_Name:
+                name = EnumerationLiteralSymbol(nodes.Get_Choice_Name(item))
+                value = GetExpressionFromNode(nodes.Get_Associated_Expr(item))
+                choices.append(NamedAggregateElement(name, value))
+            elif kind == nodes.Iir_Kind.Choice_By_Others:
+                expression = None
+                choices.append(OthersAggregateElement(expression))
```

Comment:
Choice_By_Others have a value.
I think this can be slightly refactorized: `value` can be extracted before the `if`.

Reply:
Oh, you're right. It will be fixed with the next set of enhancements.